src/rng.rs

```diff
@@ -131,8 +131,104 @@ macro_rules! rng_core {
     };
 }
 
-rng_core!(u32, u16, u8);
+// Only for types larger than 32 bits
+macro_rules! rng_core_large {
+    ($($type:ty),+) => {
+        $(
+            impl RngCore<$type> for Rng {
+                fn gen(&mut self) -> Result<$type, ErrorKind> {
+                    const WORDS: usize = mem::size_of::<$type>() / mem::size_of::<u32>();
+                    let mut res: $type = 0;
+
+                    for i in 0..WORDS {
+                        res |= (self.value()? as $type) << (i * (mem::size_of::<u32>() * 8))
+                    }
+
+                    Ok(res)
+                }
+
+                fn fill(&mut self, dest: &mut [$type]) -> Result<(), ErrorKind> {
+                    let len = dest.len() * (mem::size_of::<$type>() / mem::size_of::<u32>());
+                    let ptr = dest.as_mut_ptr() as *mut u32;
+                    let slice_u32 = unsafe { core::slice::from_raw_parts_mut(ptr, len) };
```

Comment:
Is alignment an issue?

Reply:
```
mem::align_of::<u32>() = 4
mem::align_of::<u64>() = 8
mem::align_of::<u128>() = 8
```
I don't think so.  u64/u128 is 8-byte aligned which would be valid for 4-byte aligned values as well. https://doc.rust-lang.org/reference/type-layout.html#size-and-alignment

Reply:
Oh right, I hadn't seen the distinction between `rng_core` and `rng_core_large`.